### PR TITLE
LPS-187802 [Marketplace] Implementation of the button to resend the invitation

### PR DESCRIPTION
**Pull request by juliocezar96**
[LPS-187802](https://liferay.atlassian.net/browse/LPS-187802)

### CODE REVIEW

**Comment by juliocezar96** [comment]
## :fire: GitRay Report 
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.
 ## Jira Tasks 

 <html><details><summary>More details about: <b>LPS-187802 [FE] Implementation of the button to resend the invitation</b>.</summary>
              <br />
              <table>
              <tr>
            <th>Title</th>
            <td><a target="_blank" href="https://issues.liferay.com/browse/LPS-187802">[FE] Implementation of the button to resend the invitation</a></td>
        </tr>
              <tr>
            <th>Type</th>
            <td>Technical Task</td>
        </tr>
              <tr>
            <th>Priority</th>
            <td>Low</td>
        </tr>
              <tr>
            <th>Sub-Tasks</th>
            <td>No Sub-Tasks Opened</td>
        </tr>
              <tr>
            <th>Status</th>
            <td>In Progress</td>
        </tr>
              <tr>
            <th>Assignee</th>
            <td>Julio Cezar</td>
        </tr>
              <tr>
            <th>Reporter</th>
            <td>Julio Cezar</td>
        </tr>
                </table>
          </details></html> 

---

Hey, don't forget to update all related tickets with correct status, such as **In Review** :wink: 
### Git Stats 

12 files changed, 686 insertions(+), 393 deletions(-)

**Comment by juliocezar96** [comment]
ci:test:sf

**Comment by juliocezar96** [comment]
ci:test:sf

**Comment by juliocezar96** [comment]
ci:test:sf

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/types.d.ts [hunk 1]
```diff
@@ -390,3 +390,23 @@ type UserAccount = {
 	isPublisherAccount: boolean;
 	password: string;
 };
+
+type RequestBody = {
```
Comment (joanagrego) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/types.d.ts:394]: You can try something like
```suggestion
type RequestBody = {[keys: string]: string}
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx [hunk 2]
```diff
@@ -47,32 +128,60 @@ export function MemberProfile({member, setSelectedMember}: MemberProfileProps) {
...
+
+						<button className="member-profile-button-edit-member">
+							<span className="member-profile-button-edit-member-label-edit">
+								Edit
+							</span>
+
+							<span className="icon-container-angle-down-small">
+								<ClayIcon symbol="angle-down-small" />
+							</span>
+						</button>
```
Comment (joanagrego) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx:181]:
```suggestion
				{userLogged?.isAdminAccount && (
				
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.scss [hunk 3]
```diff
@@ -1,9 +1,66 @@
...
+		background-color: #0053f0;
+		border: none;
+		border-radius: 0.5rem;
+		color: #fff;
+		font-size: 1rem;
+		font-weight: 600;
+		height: 2.5rem;
+		line-height: 1.5rem;
+		margin-right: 1rem;
+		width: max-content;
```
Comment (joanagrego) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.scss:20]: Try to use bootstrap here

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/utils/api.ts [hunk 4]
```diff
@@ -941,3 +941,41 @@ export async function updateProductSpecification({
...
+		`${baseURL}/o/c/useradditionalinfos/?filter=r_userToUserAddInfo_userId eq '${userId}' and contains(sendType,'shipping')`,
+		{headers}
+	);
+
+	const response = await userAdditionalInfos.json();
+
+	if (!response.acceptInviteStatus) {
+		return response;
+	}
+}
```
Comment (joanagrego) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/utils/api.ts:968]: You have two API call functions with this name. Try to turn it more generic using just one for both cases.

**Comment by juliocezar96** [comment]
ci:test:sf

**Comment by juliocezar96** [comment]
ci:test:sf

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx [hunk 5]
```diff
@@ -20,25 +21,105 @@ import './MemberProfile.scss';
...
-export function MemberProfile({member, setSelectedMember}: MemberProfileProps) {
+export function MemberProfile({
+	memberUser,
+	renderToast,
+	setSelectedMember,
+	userLogged,
+}: MemberProfileProps) {
 	const [{gravatarAPI}, _] = useAppContext();
 
+	const handlePut = async (event: React.FormEvent) => {
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx:56]: Hey @juliocezar96 can we choose a better function name here? You can use something that describes what this variable is doing

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx [hunk 6]
```diff
@@ -20,25 +21,105 @@ import './MemberProfile.scss';
...
+		event.preventDefault();
+
+		const myUserAdditionalInfos = await getMyUserAditionalInfos(
+			memberUser.userId
+		);
+		const newPassword = createPassword();
+
+		for (const userAdditionInfo of myUserAdditionalInfos.items || []) {
+			const response = await updateUserAdditionalinfos(
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx:65]:
```suggestion
			const updatedUserInfos = await updateUserAdditionalinfos(
```
You can use a more descriptive name for this variable

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx [hunk 7]
```diff
@@ -20,25 +21,105 @@ import './MemberProfile.scss';
...
+				userAdditionInfo.id
+			);
+
+			if (response.sendType.key === 'canceled') {
+				await updateUserPassword(newPassword, memberUser.userId);
+
+				const roles = await getAccountRolesOnAPI(
+					response.r_accountEntryToUserAdditionalInfo_accountEntryId
+				);
+				const accountsRole = roles?.find(
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx:76]:
```suggestion
				const accountRoles = roles?.find(
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx [hunk 8]
```diff
@@ -20,25 +21,105 @@ import './MemberProfile.scss';
...
+					(accountRole: AccountRole) =>
+						accountRole.name === 'Invited Member'
+				);
+
+				await callRolesApi(
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx:81]: What information does this request fetch? You can change the name to something more descriptive

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/MembersPage/MembersPage.tsx [hunk 9]
```diff
@@ -235,8 +238,10 @@ export function MembersPage({
 				>
 					{selectedMember ? (
 						<MemberProfile
-							member={selectedMember}
+							memberUser={selectedMember}
+							renderToast={renderToast}
 							setSelectedMember={setSelectedMember}
+							userLogged={userLogged}
 						></MemberProfile>
```
Comment (welitonpe) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/pages/MembersPage/MembersPage.tsx:245]: You can close directly the component
```suggestion
						/>
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx [hunk 10]
```diff
@@ -20,25 +21,105 @@ import './MemberProfile.scss';
...
+					response.r_accountEntryToUserAdditionalInfo_accountEntryId,
+					accountsRole.id,
+					response.r_userToUserAddInfo_userId
+				);
+
+				const newInvite = await addAdditionalInfo({
+					acceptInviteStatus: false,
+					accountGroupERC: response.accountGroupERC,
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx:89]:
```suggestion
					accountGroupERC: updatedUserInfo.accountGroupERC,
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx [hunk 11]
```diff
@@ -20,25 +21,105 @@ import './MemberProfile.scss';
...
+					accountName: response.accountName,
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx:90]:
```suggestion
					accountName: updatedUserInfo.accountName,
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx [hunk 12]
```diff
@@ -20,25 +21,105 @@ import './MemberProfile.scss';
...
+					emailOfMember: response.emailOfMember,
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx:91]:
```suggestion
					emailOfMember: updatedUserInfo.emailOfMember,
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx [hunk 13]
```diff
@@ -20,25 +21,105 @@ import './MemberProfile.scss';
...
+					inviteURL:
+						Liferay.ThemeDisplay.getPortalURL() +
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx:93]: I believe you can save this information in a variable to use here

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx [hunk 14]
```diff
@@ -20,25 +21,105 @@ import './MemberProfile.scss';
...
+						'/c/login?redirect=' +
+						getSiteURL() +
+						'/loading',
+					inviterName: response.inviterName,
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx:97]:
```suggestion
					inviterName: updatedUserInfo.inviterName,
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx [hunk 15]
```diff
@@ -20,25 +21,105 @@ import './MemberProfile.scss';
...
+					mothersName: newPassword,
+					r_accountEntryToUserAdditionalInfo_accountEntryId:
+						response.r_accountEntryToUserAdditionalInfo_accountEntryId,
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx:100]:
```suggestion
						updatedUserInfo.r_accountEntryToUserAdditionalInfo_accountEntryId,
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx [hunk 16]
```diff
@@ -20,25 +21,105 @@ import './MemberProfile.scss';
...
+					r_userToUserAddInfo_userId:
+						response.r_userToUserAddInfo_userId,
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx:102]:
```suggestion
						updatedUserInfo.r_userToUserAddInfo_userId,
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx [hunk 17]
```diff
@@ -20,25 +21,105 @@ import './MemberProfile.scss';
...
+					roles: response.roles,
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx:103]:
```suggestion
					roles: updatedUserInfo.roles,
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx [hunk 18]
```diff
@@ -20,25 +21,105 @@ import './MemberProfile.scss';
...
+					sendType: {key: 'shipping', name: 'Shipping'},
+					userFirstName: response.userFirstName,
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx:105]:
```suggestion
					userFirstName: updatedUserInfo.userFirstName,
```

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx [hunk 19]
```diff
@@ -47,31 +128,59 @@ export function MemberProfile({member, setSelectedMember}: MemberProfileProps) {
...
+							{memberUser.email}, Never Logged In
+						</div>
+					)}
+				</div>
+
+				{userLogged?.isAdminAccount && (
+					<div className="member-profile-resend-invitation ml-auto">
+						<button
+							className="h-50 member-profile-button-resend-invitation mr-3"
+							onClick={(event) => handlePut(event)}
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/components/MemberProfile/MemberProfile.tsx:167]: Change the name of this function

**Review by welitonpe — COMMENTED** [review]
Good Job :fire:

**Review by welitonpe — APPROVED** [review]

File: workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/utils/api.ts [hunk 20]
```diff
@@ -941,3 +941,41 @@ export async function updateProductSpecification({
 
 	return await response.json();
 }
+export async function updateUserAdditionalinfos(body: Object, id: number) {
```
Comment (claricebrasil) [inline workspaces/liferay-marketplace-workspace/client-extensions/liferay-marketplace-custom-element/src/utils/api.ts:944]:
```suggestion
export async function updateUserAdditionalInfos(body: Object, id: number) {
```

**Review by claricebrasil — CHANGES_REQUESTED** [review]

**Comment by joanagrego** [comment]
ci:forward

**Comment by joanagrego** [comment]
ci:stop

**Review by claricebrasil — APPROVED** [review]

**Comment by juliocezar96** [comment]
ci:forward

**Comment by juliocezar96** [comment]
ci:forward:force

**Comment by juliocezar96** [comment]
ci:stop

**Comment by juliocezar96** [comment]
ci:test:sf

**Comment by juliocezar96** [comment]
ci:test:sf

**Comment by juliocezar96** [comment]
ci:forward:force

**Comment by juliocezar96** [comment]
ci:forward:force

**Comment by juliocezar96** [comment]
ci:forward:force

**Pull request closed without merging**